src/LodeRunner.API.Test/IntegrationTests/ExecutingTestRun/TestRunExecution.cs

```diff
@@ -0,0 +1,283 @@
+                    testRunPayload.SetMockDataToLoadTestLodeRunnerApi(testRunName, clientStatusId, apiListeningOnPort, loadClientCount);
+
+                    HttpResponseMessage postedResponse = await httpClient.PostEntity<TestRun, TestRunPayload>(testRunPayload, TestRunsUri, this.output);
+                    Assert.Equal(HttpStatusCode.Created, postedResponse.StatusCode);
+
+                    // Validate Test Run Entity
+                    var postedTestRun = await postedResponse.Content.ReadFromJsonAsync<TestRun>(this.jsonOptions);
+                    var gottenHttpResponse = await httpClient.GetItemById<TestRun>(TestRunsUri, postedTestRun.Id, this.output);
+
+                    Assert.Equal(HttpStatusCode.OK, gottenHttpResponse.StatusCode);
+                    var gottenTestRun = await gottenHttpResponse.Content.ReadFromJsonAsync<TestRun>(this.jsonOptions);
+
+                    Assert.Equal(JsonSerializer.Serialize(postedTestRun), JsonSerializer.Serialize(gottenTestRun));
+
+                    gottenTestRunId = gottenTestRun.Id;
+
+                    // Attempt to get TestRun for N retries or until condition has met.
+                    (HttpStatusCode testRunStatusCode, TestRun readyTestRun) = await httpClient.GetEntityByIdRetries<TestRun>(TestRunsUri, postedTestRun.Id, this.jsonOptions, this.output, this.ValidateCompletedTime, 10, 2000);
+
+                    // Validate results
+                    Assert.Equal(HttpStatusCode.OK, testRunStatusCode);
+                    Assert.True(readyTestRun.CompletedTime != null, "CompletedTime is null.");
+                    Assert.True(readyTestRun.LoadClients.Count == loadClientCount, $"LoadClients.Count do not match the expected value [{loadClientCount}]");
+                    Assert.True(readyTestRun.ClientResults.Count == readyTestRun.LoadClients.Count, "ClientResults.Count do not match LoadClients.Count");
+                    var clientResult = readyTestRun.ClientResults[0];
+                    Assert.True(clientResult.TotalRequests == clientResult.FailedRequests + clientResult.SuccessfulRequests, $"TotalRequests {clientResult.TotalRequests} does not match expected value {clientResult.FailedRequests + clientResult.SuccessfulRequests}");
+
+                    this.output.WriteLine($"TestRun passed validation.");
+
+                    // Verify that ClientStatus is Ready again.
+                    await this.VerifyLodeRunnerClientStatusIsReady(httpClient, clientStatusId);
+
+                    // End LodeRunner Context.
+                    lodeRunnerAppContext.End();
+                    this.output.WriteLine($"Stopping LodeRunner Application (client mode) [ClientStatusId: {clientStatusId}]");
+
+                    lodeRunnerAPIContext.End();
+                    this.output.WriteLine($"Stopping LodeRunner API.");
+                }
+                else
+                {
+                    Assert.True(false, "Unable to start LodeRunner App Context.");
```

Comment:
Instead of an if else for the assert, what do you think of something like an 

Assert.True(lodeRunnerAppContext.Start(), "Unable to start LodeRunner App Context.");
Assert.True(lodeRunnerAPIContext.Start(), "Unable to start LodeRunner API Context.");

Then you wouldn't need the logic in the if-else since it won't progress if the Asserts fail

Reply:
yes, I think that could work as well, done.